pandas/_libs/lib.pyx

```diff
@@ -2160,11 +2195,25 @@ def maybe_convert_numeric(
     if seen.check_uint64_conflict():
         return values
 
+    # This occurs since we disabled float nulls showing as null in anticipation
+    # of seeing ints that were never seen. So then, we return float
+    if convert_to_nullable_integer and seen.null_ and not seen.int_:
+        seen.float_ = True
+
     if seen.complex_:
         return complexes
     elif seen.float_:
+        if seen.null_ and convert_to_floating_array:
+            from pandas.core.arrays import FloatingArray
+            return FloatingArray(floats, mask.view(np.bool_))
         return floats
     elif seen.int_:
+        if seen.null_ and convert_to_nullable_integer:
+            from pandas.core.arrays import IntegerArray
```

Comment:
same as for the bool conversion, can this be done in base_parser.py?

Reply:
It's more convenient this way, since I would have to special-case in base_parser. In the future, I think pd.to_numeric might use this. Is the problem with circular dependencies?

Reply:
> Is the problem with circular dependencies?

More or less, yes.  Right now we can test maybe_convert_numeric in isolation and a relatively simple mental model.  My preference is to keep it that way.